### ISSUE TITLE
Features: Browse folders, FLAC/cue format and other audio formats

### ISSUE THREAD

**Issue by BenMcLean** [basings/selfhosted-music-overview#186]
A few missing features have prevented me from using any music streaming server app on my massive multi-terabyte music collection:

1. Browse folders: My collection is already organized, named and tagged how I want it. The app should just browse folders. It doesn't need a database of all the music or even to know how much music is there: it only needs read-only access to the folders where my music is stored and that's all. Apps come and go but my collection is forever.

2. FLAC/cue support: I like to rip my music CDs using ExactAudioCopy to have one FLAC file for the disc and a cuesheet for the tracks. This is a valid way to store music and should be supported.

3. Some of my music is in other formats like APE and WavPack. I'd need the music server program to support those formats.

To my knowledge ***no music streaming server program exists with these features***, so I choose none. I've connected via SFTP with Kodi before and that kind of works a little but I'm not using any dedicated music streaming server.

Even though no music server exists for these features, I'd like to see a red "X" in a column for all of the servers that exist for not supporting these features.

**Comment by basings** [comment]
Have you checked all of the services? If so, do you want to open a pr?

**Comment by BenMcLean** [comment]
I'd have to check each one to be sure, but i'm pretty sure since I've looked at many.

But yeah, if this project is open to that, I could a PR.

**Comment by basings** [comment]
nextcloud music should be able to browse folders? https://apps.nextcloud.com/apps/music
2.according to 2 forum entries I found logitech media server supports that

**Comment by basings** [comment]
At least navidrome supports wavpack

**Comment by BenMcLean** [comment]
> nextcloud music should be able to browse folders? https://apps.nextcloud.com/apps/music 2.according to 2 forum entries I found logitech media server supports that

Yeah but those are severely limited in the formats they support and absolutely do not support FLAC/cue.

**Comment by basings** [comment]
On the issue tracker it reads as if it was supported https://github.com/Logitech/slimserver/issues/277 and I can't verify it as I don't have the server, I trust you on this 👍🏻

**Comment by BenMcLean** [comment]
> On the issue tracker it reads as if it was supported [Logitech/slimserver#277](https://github.com/Logitech/slimserver/issues/277) and I can't verify it as I don't have the server, I trust you on this 👍🏻

I stopped even looking at their app when I read [this](https://wiki.slimdevices.com/index.php/Connecting_remotely.html):
> NOTE: exposing your Logitech Music Server to the internet is a bad idea. This page is left here for reference. But we strongly suggest you don't do this. Use a VPN if you want to access your music from the outside.

So their app is specifically for people using Logitech hardware devices on a LAN. Obviously, I'm looking for something general purpose that works over the Internet.

**Comment by basings** [comment]
That's if you don't use SSL. It's described further down the page. 
They state that it's a security risk because of the port opening.

I'm not sure why they write it as if the server was inherently insecure.

**Comment by BenMcLean** [comment]
> That's if you don't use SSL. It's described further down the page. They state that it's a security risk because of the port opening.
> 
> I'm not sure why they write it as if the server was inherently insecure.

Oh ... that might make their app worth looking at after all then! Thanks!

**Comment by BenMcLean** [comment]
Looking into Logitech Media Server a bit more, it seems that while there are clients for Android to listen to LMS, it's designed to have one person listen at a time for each server. It isn't like Jellyfin where you can have multiple users consuming different content at the same time. That's more what I was thinking of but LMS with one user might be better than nothing.

**Comment by BenMcLean** [comment]
I am looking into writing my own music server app because none of the existing apps do this stuff.

**Comment by BenMcLean** [comment]
> I am looking into writing my own music server app because none of the existing apps do this stuff. LMS doesn't either: not really.

After having researched this, I've reached the conclusion that it's not feasible for a solo developer with no guidance to implement an HLS streaming server such as this would require. Apps like Emby/Jellyfin, LMS or navidrome managed to do this because they were made by people who already had years of experience in the technical details of multimedia playback. I can't just pull in some libraries and make this work because it's never been made in a reusable or modular way and by the time I finished anything even semi-working, web and video standards would have moved on to some new totally incompatible nonsense.
